spec/requests/contact_spec.rb

```diff
@@ -0,0 +1,162 @@
+require 'rails_helper'
+
+def check_attributes(parsed_contact, target_contact)
+  expect(parsed_contact[:name]).to eq(target_contact[:name])
+  expect(parsed_contact[:email]).to eq(target_contact[:email])
+  expect(parsed_contact[:city]).to eq(target_contact[:city])
+  expect(parsed_contact[:birthday]).to eq(target_contact[:birthday])
+end
+
+RSpec.describe :Contact, type: :request do
+  describe 'GET /index' do
+    subject(:fetch_contacts) do
+      get contacts_path, headers: headers
+    end
```

Comment:
```suggestion
    before { get contacts_path, headers: headers }
```

Ai sempre antes de rodar os testes vai fazer a request, e nao precisa chamar dentro do `it`.

Pergunta, por que precisa das headers aqui?

Reply:
Na verdade não precisa, peguei outro exemplo de base e acabou colocando isso sem necessidade.